base/strings/basic.jl

```diff
@@ -512,7 +512,7 @@ isascii(s::AbstractString) = all(isascii, s)
 ## string map, filter, has ##
 
 function map(f, s::AbstractString)
-    out = IOBuffer(StringVector(sizeof(s)), true, true)
+    out = IOBuffer(StringVector(sizeof(s)), read=true, write=true)
```

Comment:
Passing a `StringVector` is pretty common here.  It would be good to add a `sizehint` keyword argument so that you could just do `IOBuffer(sizehint=sizeof(s))` here and in similar places.

Reply:
Okay, I'll take care of it.